### PR TITLE
Fix (scripts/Scholomance) - Kirtonos the Herald event now starts correctly again

### PR DESCRIPTION
**Pull request by patou01**
The root cause of the issue is that when a creature is summoned by an instance, the ``IsSummonedBy()`` method is not called anymore. 

Additionally, I noticed that he's swimming through the air, so I checked how it's done for Onyxia and tried to copy from there.

## Changes Proposed:
-  Use `DoAction` on the summoned creature (Kirtonos). Reuse the `IN_PROGRESS` define because it doens't hurt..
- Add NO_MOVE_FLAGS_UPDATE, I copied the query from Keira3, I don't recall how they're supposed to be made if it should OR or actually use absolute values.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/14944

## Tests Performed:

- Ran it on my own computer, the boss seems to start moving and not swim in the air in humanoid form.
-  I did notice earlier on that he doesn't always start, so maybe we want to have a few tries to be sure.

## How to Test the Changes:

1. .teleport scholomance
2. Enter the dungeon
3. .go xyz 325.122009 93.6436000 101.637001
4. .add 13523
5. right click the brazier
6. Kirtonos should spawn and start moving (flying around) and then come to the place where you are. He should behave normally after that, no swimming in the air.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/Scholomance/boss_kirtonos_the_herald.cpp [hunk 1]
```diff
@@ -128,6 +118,25 @@ class boss_kirtonos_the_herald : public CreatureScript
             }
         }
 
+        void SetData(uint32 type, uint32 data) override
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Scholomance/boss_kirtonos_the_herald.cpp:121]: In that case, you could also use DoAction()

File: src/server/scripts/EasternKingdoms/Scholomance/instance_scholomance.cpp [hunk 2]
```diff
@@ -131,7 +131,11 @@ class instance_scholomance : public InstanceMapScript
                             // summon kirtonos and close door
                             if (_kirtonosState == NOT_STARTED)
                             {
-                                instance->SummonCreature(NPC_KIRTONOS, KirtonosSpawn);
+                                TempSummon* sum = instance->SummonCreature(NPC_KIRTONOS, KirtonosSpawn);
+                                if(sum)
+                                {
+                                    sum->AI()->SetData(DATA_KIRTONOS_THE_HERALD, IN_PROGRESS); // Set the status of the boss to start the event
+                                }
```
Comment (Nyeriah) [inline src/server/scripts/EasternKingdoms/Scholomance/instance_scholomance.cpp:138]:
```suggestion
                                if (Creature* kirtonos = instance->SummonCreature(NPC_KIRTONOS, KirtonosSpawn))
                                {
                                    kirtonos ->AI()->SetData(DATA_KIRTONOS_THE_HERALD, IN_PROGRESS); // Set the status of the boss to start the event
                                }
```

File: src/server/scripts/EasternKingdoms/Scholomance/boss_kirtonos_the_herald.cpp (hunk 1 above)
Comment (patou01) [reply src/server/scripts/EasternKingdoms/Scholomance/boss_kirtonos_the_herald.cpp]: Done, didn't know it existed.

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
Thanks for the PR 👍

**Pull request merged**